internal/service/infra_mgmt/api_hosts_data_source.go

```diff
@@ -91,32 +106,46 @@ func (d *HostsDataSource) Read(ctx context.Context, req datasource.ReadRequest,
 
 	// Read Terraform prior state data into the model
 	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	allResults, err := utils.ReadWithPages(func(offset, limit int32) ([]infra_mgmt.InfraHost, error) {
-		apiRes, _, err := d.client.InfraManagementAPI.
-			HostsAPI.
-			HostsList(ctx).
-			Filter(flex.ExpandFrameworkMapFilterString(ctx, data.Filters, &resp.Diagnostics)).
-			Tfilter(flex.ExpandFrameworkMapFilterString(ctx, data.TagFilters, &resp.Diagnostics)).
-			Offset(offset).
-			Limit(limit).
-			Execute()
+	readTimeout, diags := data.Timeouts.Read(ctx, 20*time.Minute)
```

Comment:
So we won't be using the `timeouts` defined in the TF file?

Reply:
Yes, `data.Timeouts.Read` will have the value that was configured by the user.